cmd/flarectl/dns.go

```diff
@@ -114,12 +114,15 @@ func dnsCreateOrUpdate(c *cli.Context) error {
 		}
 	} else {
 		// Record doesn't exist - create it
-		rr := cloudflare.CreateDNSRecordParams{}
-		rr.Type = rtype
-		rr.Content = content
-		rr.TTL = ttl
-		rr.Proxied = &proxy
-		rr.Priority = &priority
+		rr := cloudflare.CreateDNSRecordParams{
+			Name:     name,
```

Comment:
Changed the initialisation to make it match the `dnsCreate` method.  The key change to fix the bug is line 118, now because the `name` was not being sent to `create` which would create a ` DNS Validation Error (1004)`